api/origin_trials_api.py

```diff
@@ -33,3 +37,76 @@ def do_get(self, **kwargs):
       self.abort(500, 'Malformed response from origin trials API')
 
     return trials_list
+
+  def _validate_extension_args(
+        self, feature_id: int, ot_stage: Stage, extension_stage: Stage) -> None:
+    """Abort if any arguments used for origin trial extension are invalid."""
+    # The stage should belong to the feature.
+    if feature_id != extension_stage.feature_id:
+      self.abort(400, ('Stage does not belong to feature. '
+                       f'feature_id: {feature_id}, '
+                       f'stage_id: {extension_stage.key.integer_id()}'))
+
+    trial_id = ot_stage.origin_trial_id
+    if trial_id is None:
+      self.abort(400, f'Invalid argument for trial_id: {trial_id}')
+
+    milestones = extension_stage.milestones
+    if milestones is None or milestones.desktop_last is None:
+      self.abort(404, f'Extension stage has no end milestone defined.')
+    intent_url = extension_stage.intent_thread_url
+    if intent_url is None or not validators.url(intent_url):
+      self.abort(400, ('Invalid argument for extension_intent_url: '
+                       f'{intent_url}'))
+
+    gate = Gate.query(Gate.stage_id == extension_stage.key.integer_id()).get()
+    if not gate or gate.state != Vote.APPROVED:
+      self.abort(400, 'Extension has not received the required approvals.')
+
+  def do_post(self, **kwargs):
+    """Extends an existing origin trial"""
+    feature_id = int(kwargs['feature_id'])
+    extension_stage_id = int(kwargs['extension_stage_id'])
+    # Check that feature ID is valid.
+    if feature_id is None or feature_id == 0:
```

Comment:
`if not feature_id:` would be more pythonic.  The same change could be made again below.

Reply:
Agreed. Updated!